### ISSUE TITLE
[BUG][TypeScript fetch] "oneOf" in query parameter generates wrong code

### ISSUE THREAD

**Issue by targos** [OpenAPITools/openapi-generator#12510]
#### Bug Report Checklist

- [x] Have you provided a full/minimal spec to reproduce the issue?
- [x] Have you validated the input using an OpenAPI validator ([example](https://apidevtools.org/swagger-parser/online/))?
- [ ] Have you [tested with the latest master](https://github.com/OpenAPITools/openapi-generator/wiki/FAQ#how-to-test-with-the-latest-master-of-openapi-generator) to confirm the issue still exists?
- [x] Have you searched for related issues/PRs?
- [x] What's the actual output vs expected output?
- [ ] [Optional] Sponsorship to speed up the bug fix or feature request ([example](https://github.com/OpenAPITools/openapi-generator/issues/6178))

##### Description

In version 6.0.0, if `oneOf` is used in the schema for a query parameter, the generated code is wrong in a few ways:
- The file which contains the type definition for the parameter also has code referencing functions that do not exist:
![image](https://user-images.githubusercontent.com/2352663/171592234-11a83892-28ae-4d81-96ee-44a94f5d17e5.png)
- The interface generated for the query parameters uses the type from that file but doesn't import it:
![image](https://user-images.githubusercontent.com/2352663/171592411-49c92536-74b6-49ab-aeb7-449576914fbb.png)

##### openapi-generator version

6.0.0. This is a regression from version 5.4.0.

##### OpenAPI declaration file content or url

https://gist.github.com/targos/ce0dcf58be520f94c1bdab868f14e17b

##### Generation Details

```json
{
  "$schema": "node_modules/@openapitools/openapi-generator-cli/config.schema.json",
  "spaces": 2,
  "generator-cli": {
    "version": "6.0.0",
    "generators": {
      "typescript-fetch": {
        "generatorName": "typescript-fetch",
        "glob": "openapi.json",
        "output": "test/api",
        "additionalProperties": {
          "enumPropertyNaming": "UPPERCASE"
        }
      }
    }
  }
}
```

##### Steps to reproduce

```
npm install -D @openapitools/openapi-generator-cli
npx openapi-generator-cli generate
```

**Comment by frederikprijck** [comment]
👋 

Hello @macjohnny and @Nukesor.
Sorry to revive this, but I just ran the latest master against the provided spec file using

```sh
docker run --rm -v "${PWD}:/local" openapitools/openapi-generator-cli generate \
    -i https://gist.githubusercontent.com/targos/ce0dcf58be520f94c1bdab868f14e17b/raw/85774f17d9bd349172de3abb1c1ed5bcc1748f72/openapi.json \
    -g typescript-fetch \
    -o /local/out/typescript-fetch
```

But I still get the same errors. I also tried this with `6.0.1` locally, which should be the version that contains the fix #12513.

<img width="1183" alt="image" src="https://user-images.githubusercontent.com/2146903/220120370-74ebca94-4b77-42c3-a7ef-feb24f748ba2.png">

Is this supposed to be fixed, or am I missing something?

**Comment by Nukesor** [comment]
Hi :) 

I actually stopped use this feature, as another language in our stack didn't support this feature until recently.
However, I might revisit this in the near future.

**Comment by xseman** [comment]
@frederikprijck @macjohnny I'm having same issiue using typescript-fetch `v7.0.1`

> [oneOf](https://swagger.io/docs/specification/data-models/oneof-anyof-allof-not/#oneof) – validates the value against exactly one of the subschemas

Generated logic isn't working as expected for `exactly one of the subschemas`. Is this due to `v6.0.1` implementation or just a bug? If more work is needed, I'm happy to help.

**Comment by macjohnny** [comment]
@xseman help is always welcome!

**Comment by Wei102193** [comment]
how come this still has not been fixed tho lol

**Comment by denisvanmorgan** [comment]
@macjohnny hi, just a friendly ping. is there any chance you will look into this in the nearest future? the bug still persists and it looks like there is no other way to fix it other than move to v5.4.0 which is not desirable. It's also described in these issues #14763 #14806 #14611 . They seem very related to each other

**Comment by macjohnny** [comment]
@denisvanmorgan i am happy to review your PR
